lib/payload/cloak.vala

```diff
@@ -99,23 +99,27 @@ namespace Frida {
 			listeners.add (readdir_listener);
 			interceptor.attach (readdir, readdir_listener);
 
-			var readdir64 = Gum.Module.find_export_by_name (libc, "readdir64");
-			if (readdir64 != null && readdir64 != readdir) {
+			/*
+			 * It seems that when using the musl CRT, the compiler incorrectly determines the symbols 'readdir' and 'readdir64' collide
+			 * resulting in a redefinition error, perhaps as a result of an errant #define. Hence the name has been slightly changed.
+			 */
+			var readdir64_impl = Gum.Module.find_export_by_name (libc, "readdir64");
+			if (readdir64_impl != null && readdir64_impl != readdir) {
 				var listener = new ReadDirListener (this, MODERN);
 				listeners.add (listener);
-				interceptor.attach (readdir64, listener);
+				interceptor.attach (readdir64_impl, listener);
 			}
 
 			var readdir_r = Gum.Module.find_export_by_name (libc, "readdir_r");
 			var readdir_r_listener = new ReadDirRListener (this, LEGACY);
 			listeners.add (readdir_r_listener);
 			interceptor.attach (readdir_r, readdir_r_listener);
 
-			var readdir64_r = Gum.Module.find_export_by_name (libc, "readdir64_r");
-			if (readdir64_r != null && readdir64_r != readdir_r) {
+			var readdir64_r_impl = Gum.Module.find_export_by_name (libc, "readdir64_r");
+			if (readdir64_r_impl != null && readdir64_r_impl != readdir_r) {
 				var listener = new ReadDirRListener (this, MODERN);
 				listeners.add (listener);
-				interceptor.attach (readdir64_r, listener);
+				interceptor.attach (readdir64_r_impl, listener);
 			}
```

Comment:
I think we should also suffix the other names for consistency. I'm not sure we need the comment though, as it's generally a bad idea to name variables the same as standard functions.